src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/EditContactDetailsModal/EditContactDetailsModal.tsx

```diff
@@ -55,72 +58,141 @@ const PrimaryContactIcon = styled(BookmarkIcon)(({ theme }) => ({
   color: theme.palette.cruGrayMedium.main,
 }));
 
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
 interface EditContactDetailsModalProps {
   contact: ContactDetailsTabQuery['contact'];
+  accountListId: string;
   isOpen: boolean;
-  handleOpenModal: (open: boolean) => void;
+  handleClose: () => void;
 }
+
 export const EditContactDetailsModal: React.FC<EditContactDetailsModalProps> = ({
+  accountListId,
   contact,
   isOpen,
-  handleOpenModal,
+  handleClose,
 }): ReactElement<EditContactDetailsModalProps> => {
   const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const [
+    updateContact,
+    { loading: updating },
+  ] = useUpdateContactDetailsMutation();
+
+  const contactSchema: yup.SchemaOf<
+    Pick<ContactUpdateInput, 'name' | 'id'>
+  > = yup.object({
+    name: yup.string().required(),
+    id: yup.string().required(),
+  });
+
+  // TODO Add logic to update primary person when it comes available in mutation
+  const onSubmit = async (attributes: ContactUpdateInput) => {
+    try {
+      await updateContact({
+        variables: {
+          accountListId,
+          attributes: {
+            name: attributes.name,
+            id: attributes.id,
+          },
+        },
+      });
+      enqueueSnackbar(t('Contact updated successfully'), {
+        variant: 'success',
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
+  };
 
   return (
     <Modal
       isOpen={isOpen}
       title={t('Edit Contact Details')}
-      handleClose={() => handleOpenModal(false)}
+      handleClose={handleClose}
     >
-      <DialogContent dividers>
-        <ContactEditContainer>
-          <ContactInputWrapper>
-            <ContactInputField
-              label={t('Contact')}
-              value={contact.name}
-              fullWidth
-            />
-          </ContactInputWrapper>
-          <ContactInputWrapper>
-            <PrimaryContactIcon />
+      <Formik
```

Comment:
This library looks neat, I'll have to do more forms later and this looks like it has potential to make it easier in some ways. The validation piece looks nice!

Reply:
Yeah, Tatai chose Formik originally, but it has been great to work with! I've been trying to learn all I can about it and best practices as I look towards the future and the eventual ERT react rewrite. There's a lot of form validation in that app haha.